### PR TITLE
docs: add missing locators + improvements

### PR DESCRIPTION
**Pull request by debs-obrien**

### CODE REVIEW

File: docs/src/locators.md [hunk 1]
```diff
@@ -168,10 +192,112 @@ var locator = page.FrameLocator("#my-frame").GetByText("Submit");
...
+```python sync
+page.get_by_label("Password").fill("secret")
+```
+
+```csharp
+await page.GetByLabel("Password").FillAsync("secret");
+```
+
+### Locate by placeholder text with getByPlaceholder
+[`method: Page.getByPlaceholder`]
```
Comment (dgozman) [inline docs/src/locators.md:266]: This method link is out of place.

File: docs/src/locators.md [hunk 2]
```diff
@@ -168,10 +192,112 @@ var locator = page.FrameLocator("#my-frame").GetByText("Submit");
...
+
+Inputs may have a placeholder attribute to hint to the user what value should be entered. You can locate the placeholder using [`method: Page.getByPlaceholder`].
```
Comment (dgozman) [inline docs/src/locators.md:268]:
```suggestion
Inputs may have a placeholder attribute to hint to the user what value should be entered. You can locate such an input using [`method: Page.getByPlaceholder`].
```

File: docs/src/locators.md [hunk 3]
```diff
@@ -168,10 +192,112 @@ var locator = page.FrameLocator("#my-frame").GetByText("Submit");
...
+page.get_by_placeholder("name@example.com").fill("playwright@microsoft.com")
+```
+
+```csharp
+await page.GetByPlacheolder("name@example.com").FillAsync("playwright@microsoft.com");
+```
+
+### Locate by text using getByText
+
+The easiest way to find an element is to look for the text it contains. You can match by a substring, exact string, or a regular expression when using [`method: Page.getByText`]
```
Comment (dgozman) [inline docs/src/locators.md:300]: nit: missing punctuation at the end.

File: docs/src/locators.md [hunk 4]
```diff
@@ -221,106 +347,101 @@ await page.GetByTestId("product-item").Filter(new() { HasText = "Playwright Book
...
 :::
 
-### Locate based on accessible attributes with [`method: Page.getByRole`]
-
-The [`method: Page.getByRole`] locator reflects how users and assistive technology percieve the page, for example whether some element is a button or a checkbox. When locating by role, you should usually pass the accessible name as well, so that locator pinpoints the exact element.
+### Locate by alt text with getByAltText
 
-```js
-await page.getByRole('button', { name: /submit/i }).click();
+All images should have a valid alt tag which describes the image. You can locate the placeholder using [`method: Page.getByPlaceholder`].
```
Comment (dgozman) [inline docs/src/locators.md:352]:
```suggestion
All images should have an `alt` attribute that describes the image. You can locate an image based on the text alternative using [`method: Page.getByAltText`].
```

File: docs/src/locators.md [hunk 5]
```diff
@@ -221,106 +347,101 @@ await page.GetByTestId("product-item").Filter(new() { HasText = "Playwright Book
...
+For example, consider the following DOM structure.
 
-await page.get_by_role("checkbox", checked=True, name="Check me"]).check()
+```html
+<img alt="playwright logo" src="/playwright-logo.png" />
 ```
 
-```python sync
-page.get_by_role("button", name=re.compile("submit", re.IGNORECASE)).click()
+You can fill the input after locating it by the label text:
```
Comment (dgozman) [inline docs/src/locators.md:361]:
```suggestion
You can click on the image after locating it by the text alternative:
```

File: docs/src/locators.md [hunk 6]
```diff
@@ -221,106 +347,101 @@ await page.GetByTestId("product-item").Filter(new() { HasText = "Playwright Book
...
 ```
 
-Role locators follow W3C specificaitons for [ARIA role](https://www.w3.org/TR/wai-aria-1.2/#roles), [ARIA attributes](https://www.w3.org/TR/wai-aria-1.2/#aria-attributes) and [accessible name](https://w3c.github.io/accname/#dfn-accessible-name).
+```python sync
+page.get_by_alt_text("playwright logo").click()
+```
 
-Note that role locators **do not replace** accessibility audits and conformance tests, but rather give early feedback about the ARIA guidelines.
+```csharp
+await page.getByAltText("playwright logo").click();
```
Comment (dgozman) [inline docs/src/locators.md:380]:
```suggestion
await page.GetByAltText("playwright logo").ClickAsync();
```

File: docs/src/locators.md [hunk 7]
```diff
@@ -221,106 +347,101 @@ await page.GetByTestId("product-item").Filter(new() { HasText = "Playwright Book
...
+```
+### Locate by title with getByTitle
+[`method: Page.getByTitle`]
```
Comment (dgozman) [inline docs/src/locators.md:383]: This method link is out of place.

File: docs/src/locators.md [hunk 8]
```diff
@@ -221,106 +347,101 @@ await page.GetByTestId("product-item").Filter(new() { HasText = "Playwright Book
...
 
-### Define explicit contract and use [`method: Page.getByTestId`]
+Locate an element with a matching title attribute using [`method: Page.getByTitle`].
 
-User-facing attributes like text or accessible name can change over time. In this case it is convenient to define explicit test ids.
+For example, consider the following DOM structure.
 
 ```html
-<button data-testid="directions">Itinéraire</button>
+<span title='Close'></span>
```
Comment (dgozman) [inline docs/src/locators.md:390]: I think the better example would be:

```html
<span title='Issues count'>25 issues</span>
```

```js
await expect(page.getByTitle('Issues count')).toHaveText('25 issues');
```

File: docs/src/locators.md [hunk 9]
```diff
@@ -221,106 +347,101 @@ await page.GetByTestId("product-item").Filter(new() { HasText = "Playwright Book
...
 
-By default, [`method: Page.getByTestId`] will locate elements baed on the `data-testid` attribute, but you can configure it in your test config or calling [`method: Selectors.setTestIdAttribute`].
-
-### Locate by label text with [`method: Page.getByLabel`]
+### Define explicit contract and use getByTestId
 
-Most form controls usually have dedicated labels that could be conveniently used to interact with the form. In this case, you can locate the control by its associated label.
-
-For example, consider the following DOM structure.
+User-facing attributes like text or accessible name can change over time. In this case it is convenient to define explicit test ids using [`method: Page.getByTestId`].
```
Comment (dgozman) [inline docs/src/locators.md:417]: You are not defining test ids using this method, you only query using this method. Perhaps we should expand on "how to define test ids" to make this clear?

File: docs/src/locators.md [hunk 10]
```diff
@@ -168,10 +192,112 @@ var locator = page.FrameLocator("#my-frame").GetByText("Submit");
...
 
-The easiest way to find an element is to look for the text it contains. You can match by a substring, exact string, or a regular expression.
+await page.GetByRole("checkbox", new() { Checked = true, Name = "Check me" }).CheckAsync();
+```
+
+Role locators follow W3C specificaitons for [ARIA role](https://www.w3.org/TR/wai-aria-1.2/#roles), [ARIA attributes](https://www.w3.org/TR/wai-aria-1.2/#aria-attributes) and [accessible name](https://w3c.github.io/accname/#dfn-accessible-name).
+
+Note that role locators **do not replace** accessibility audits and conformance tests, but rather give early feedback about the ARIA guidelines.
+
+### Locate by label text with getByLabel
```
[the remark this answers is not in the document]
Comment (debs-obrien) [reply docs/src/locators.md:233]: Aahh, now I understand why you do that. its just really ugly in the sidebar having the method there. so perhaps we can just drop it and leave as 
### Locate by label text

File: docs/src/locators.md [hunk 11]
```diff
@@ -168,10 +192,112 @@ var locator = page.FrameLocator("#my-frame").GetByText("Submit");
...
+
+```python sync
+page.get_by_placeholder("name@example.com").fill("playwright@microsoft.com")
+```
+
+```csharp
+await page.GetByPlacheolder("name@example.com").FillAsync("playwright@microsoft.com");
+```
+
+### Locate by text using getByText
```
[the remark this answers is not in the document]
Comment (debs-obrien) [reply docs/src/locators.md:298]: I think you should use getByRole before you use a getByText and getByTestId should be your last resort if you cannot get by a role or by a text. But thats just my opinion and Kent's opinion too. It's also how testing library layout their docs

File: docs/src/locators.md [hunk 12]
```diff
@@ -6,6 +6,30 @@ title: "Locators"
 [Locator]s are the central piece of Playwright's auto-waiting and retry-ability. In a nutshell, locators represent
 a way to find element(s) on the page at any moment.
 
+### Quick Guide
+
+These are the recommended built in locators and are available since version 1.27.
+
+- `page.getByRole` to locate by [ARIA role](https://www.w3.org/TR/wai-aria-1.2/#roles), [ARIA attributes](https://www.w3.org/TR/wai-aria-1.2/#aria-attributes) and [accessible name](https://w3c.github.io/accname/#dfn-accessible-name).
```
[the remark this answers is not in the document]
Comment (debs-obrien) [reply docs/src/locators.md:13]: damn. its so ugly but ok I get it

File: docs/src/locators.md (hunk 3 above)
Comment (dgozman) [reply docs/src/locators.md:312]: Still missing :)

File: docs/src/locators.md [hunk 13]
```diff
@@ -221,106 +359,100 @@ await page.GetByTestId("product-item").Filter(new() { HasText = "Playwright Book
...
 
-User-facing attributes like text or accessible name can change over time. In this case it is convenient to define explicit test ids.
+For example, consider the following DOM structure.
 
 ```html
-<button data-testid="directions">Itinéraire</button>
+<span title='Issues count'>25 issues</span>
 ```
 
+You can fill the input after locating it by the label text:
```
Comment (dgozman) [inline docs/src/locators.md:404]: This text is about something else.

File: docs/src/locators.md [hunk 14]
```diff
@@ -221,106 +359,100 @@ await page.GetByTestId("product-item").Filter(new() { HasText = "Playwright Book
...
+
 ```js
-await page.getByTestId('directions').click();
+await frame.getByTitle('Issues count').toHaveText('25 issues');
```
Comment (dgozman) [inline docs/src/locators.md:407]:
```suggestion
await expect(page.getByTitle('Issues count')).toHaveText('25 issues');
```

Same for other languages.

File: docs/src/locators.md [hunk 15]
```diff
@@ -486,10 +618,10 @@ page.getByRole('section').filter({ has: page.getByTestId('subscribe-button') })
 page.getByRole("section").filter(new Locator.FilterOptions().setHas(page.getByTestId("subscribe-button")))
 ```
 ```python async
-page.get_by_role("section"), has=page.get_by_test_id("subscribe-button"))
+page.get_by_role("section"), has=page.get_by_test_id("subscribe-button")
```
Comment (dgozman) [inline docs/src/locators.md:621]:
```suggestion
page.get_by_role("section").filter(has=page.get_by_test_id("subscribe-button"))
```

File: docs/src/locators.md [hunk 16]
```diff
@@ -486,10 +618,10 @@ page.getByRole('section').filter({ has: page.getByTestId('subscribe-button') })
...
 ```
 ```python sync
-page.get_by_role("section"), has=page.get_by_test_id("subscribe-button"))
+page.get_by_role("section"), has=page.get_by_test_id("subscribe-button")
```
Comment (dgozman) [inline docs/src/locators.md:624]:
```suggestion
page.get_by_role("section").filter(has=page.get_by_test_id("subscribe-button"))
```

File: docs/src/locators.md [hunk 17]
```diff
@@ -6,29 +6,66 @@ title: "Locators"
...
 ```python async
-locator = page.get_by_text("Submit")
-await locator.click()
+await page.get_by_label("User Name").fill("John")
+
+await page.get_by_label("Password").fill("secret-password")
+
+await page.get_by_role("button", { name: "Sign in" }).click()
+
+await page.get_by_text("Welcome, John!").toBeVisible();
```
Comment (dgozman) [inline docs/src/locators.md:48]: Missing `expect`

File: docs/src/locators.md [hunk 18]
```diff
@@ -6,29 +6,66 @@ title: "Locators"
...
 ```python sync
-locator = page.get_by_text("Submit")
-locator.click()
+page.get_by_label("User Name").fill("John")
+
+page.get_by_label("Password").fill("secret-password")
+
+page.get_by_role("button", { name: "Sign in" }).click()
+
+page.get_by_text("Welcome, John!").toBeVisible();
```
Comment (dgozman) [inline docs/src/locators.md:58]: Missing `expect`

File: docs/src/locators.md [hunk 19]
```diff
@@ -6,29 +6,66 @@ title: "Locators"
...
 ```csharp
-var locator = page.GetByText("Submit");
-await locator.ClickAsync();
+await page.GetByLabel('User Name').FillAsync('John');
+
+await page.GetByLabel('Password').FillAsync('secret-password');
+
+await page.GetByRole('button', { name: 'Sign in' }).ClickAsync();
+
+await expect(page.GetByText('Welcome, John!')).ToBeVisibleAsync();
```
Comment (dgozman) [inline docs/src/locators.md:68]: Wrong syntax.

File: docs/src/locators.md [hunk 20]
```diff
@@ -6,29 +6,66 @@ title: "Locators"
 [Locator]s are the central piece of Playwright's auto-waiting and retry-ability. In a nutshell, locators represent
 a way to find element(s) on the page at any moment.
 
+### Quick Guide
+
+These are the recommended built in locators.
+
+- [`method: Page.getByRole`] to locate by [ARIA role](https://www.w3.org/TR/wai-aria-1.2/#roles), [ARIA attributes](https://www.w3.org/TR/wai-aria-1.2/#aria-attributes) and [accessible name](https://w3c.github.io/accname/#dfn-accessible-name).
+- [`method: Page.getByText`] to locate by text content.
+- [`method: Page.getByLabel`] to locate a form control by associated label's text.
```
Comment (dgozman) [inline docs/src/locators.md:15]: Should we add links to the sections below for easy navigation? Not feeling strongly, just asking.

**Review by dgozman — APPROVED** [review]

**Pull request merged**